src/utils/es_query/from_kuery.js

```diff
@@ -17,33 +17,20 @@
  * under the License.
  */
 
-import { fromLegacyKueryExpression, fromKueryExpression, toElasticsearchQuery, nodeTypes } from '../../../kuery';
-import { documentationLinks } from '../../../documentation_links';
-import { NoLeadingWildcardsError } from '../../../kuery/errors';
-
-const queryDocs = documentationLinks.query;
+import { fromLegacyKueryExpression, fromKueryExpression, toElasticsearchQuery, nodeTypes } from '../kuery';
 
 export function buildQueryFromKuery(indexPattern, queries = [], config) {
   const allowLeadingWildcards = config.get('query:allowLeadingWildcards');
-
-  const queryASTs = queries.map((query) => {
+  const queryASTs = queries.map(query => {
     try {
       return fromKueryExpression(query.query, { allowLeadingWildcards });
-    }
-    catch (parseError) {
-      if (parseError instanceof NoLeadingWildcardsError) {
-        throw parseError;
-      }
```

Comment:
What was this condition doing and why was it removed?

Reply:
Because I got rid of the `NoLeadingWildcardsError` since it relied on `KbnError` in public. Instead, we just throw a regular error. This condition wasn't actually really necessary anyway since we end up throwing the error anyway below (unless it parses successfully in the old syntax). So in theory, the only thing that changes by removing this condition is when something would parse successfully in the old grammar but also has a leading wildcard, which I can't imagine of a scenario.